software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcessImpl.java

```diff
@@ -287,6 +299,89 @@ protected void postDriverStart() {
      */
     protected void connectSensors() {
         connectedSensors = true;
+        boolean retrieveMachineMetrics = config().get(RETRIEVE_MACHINE_METRICS);
+        if (retrieveMachineMetrics) connectMachineSensors();
+    }
+
+    /**
+     * Adds sensors returning details about the machine the process is running on.
+     * <p>
+     * The machine must be SSHable and running Linux.
+     */
+    protected void connectMachineSensors() {
+        Maybe<SshMachineLocation> location = Machines.findUniqueMachineLocation(getLocations(), SshMachineLocation.class);
+        if (location.isPresent() && location.get().getOsDetails().isLinux()) {
+            machineSensorFeed = SshFeed.builder()
+                    .entity(this)
+                    .period(Duration.THIRTY_SECONDS)
+                    .poll(SshPollConfig.forSensor(UPTIME)
+                            .command("cat /proc/uptime")
+                            .onFailureOrException(Functions.<Duration>constant(null))
+                            .onSuccess(new Function<SshPollValue, Duration>() {
+                                @Override
+                                public Duration apply(SshPollValue input) {
+                                    return Duration.seconds( Double.valueOf( Strings.getFirstWord(input.getStdout()) ) );
+                                }
+                            }))
+                    .poll(SshPollConfig.forSensor(LOAD_AVERAGE)
+                            .command("uptime")
+                            .onFailureOrException(Functions.constant(-1d))
+                            .onSuccess(new Function<SshPollValue, Double>() {
+                                @Override
+                                public Double apply(SshPollValue input) {
+                                    String loadAverage = Strings.getFirstWordAfter(input.getStdout(), "load average:").replace(",", "");
+                                    return Double.valueOf(loadAverage);
+                                }
+                            }))
+                    .poll(SshPollConfig.forSensor(CPU_USAGE)
+                            .command("cat /proc/stat")
```

Comment:
A problem with the cpu stats is that they are avarage since boot time, not since the last feed poll [ .period(Duration.THIRTY_SECONDS) ]
likely a separate issue but can we change this so it only accounts for usage since the last poll. It would require maintaining the counters as at the last poll time to calculate the difference.